crates/tabby-download/src/lib.rs

```diff
@@ -29,9 +72,10 @@ async fn download_model_impl(
     let model_info = registry.get_model_info(name);
     registry.save_model_info(name);
 
-    registry.migrate_model_path(name)?;
-    let model_path = registry.get_model_path(name);
-    if model_path.exists() {
+    registry.migrate_relative_model_path(name)?;
+
+    let model_path = registry.get_model_entry_path(name);
+    if let Some(model_path) = model_path {
```

Comment:
this implementation only checks sha256 for the first partition, which seems wrong to me.